airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -124,3 +130,153 @@ def get_dataset_events(
+        .where(*where_clause)
+    )
+    if ddrq is None:
+        raise NotFound(
+            "Queue event not found",
+            detail=f"Queue event with dag_id: `{dag_id}` and dataset uri: `{uri}` was not found",
+        )
+    queue_event = {"created_at": ddrq.created_at, "dag_id": dag_id, "uri": uri}
+    return queue_event_schema.dump(queue_event)
+
+
+@security.requires_access_dataset("DELETE")
+@provide_session
+def delete_dag_dataset_queue_event(
+    *, dag_id: str, uri: str, before: str | None = None, session: Session = NEW_SESSION
+) -> APIResponse:
+    """Delete a queued Dataset event for a DAG."""
+    dataset_id = session.scalars(select(DatasetModel.id).where(DatasetModel.uri == uri)).one_or_none()
+    where_clause = _generate_queue_event_where_clause(dag_id=dag_id, dataset_id=dataset_id, before=before)
+    delete_stmt = delete(DatasetDagRunQueue).where(*where_clause)
+    result = session.execute(delete_stmt)
+    if result.rowcount > 0:
+        return NoContent, HTTPStatus.NO_CONTENT
+    raise NotFound(
+        "Queue event not found",
+        detail=f"Queue event with dag_id: `{dag_id}` and dataset uri: `{uri}` was not found",
+    )
+
+
+@security.requires_access_dataset("GET")
+@provide_session
+def get_dag_dataset_queue_events(
+    *, dag_id: str, before: str | None = None, session: Session = NEW_SESSION
+) -> APIResponse:
+    """Get queued Dataset events for a DAG."""
+    where_clause = _generate_queue_event_where_clause(dag_id=dag_id, before=before)
+    query = (
+        select(DatasetDagRunQueue, DatasetModel.uri)
+        .join(DatasetModel, DatasetDagRunQueue.dataset_id == DatasetModel.id)
+        .where(*where_clause)
+    )
+    result = session.execute(query).all()
+    total_entries = get_query_count(query, session=session)
+    if not result:
+        raise NotFound(
+            "Queue event not found",
+            detail=f"Queue event with dag_id: `{dag_id}` was not found",
+        )
+    queue_events = [
+        QueueEvent(created_at=ddrq.created_at, dag_id=ddrq.target_dag_id, uri=uri) for ddrq, uri in result
+    ]
+    return queue_event_collection_schema.dump(
+        QueueEventCollection(queue_events=queue_events, total_entries=total_entries)
+    )
+
+
+@security.requires_access_dataset("DELETE")
+@provide_session
+def delete_dag_dataset_queue_events(
+    *, dag_id: str, before: str | None = None, session: Session = NEW_SESSION
+) -> APIResponse:
+    """Delete queued Dataset events for a DAG."""
+    where_clause = _generate_queue_event_where_clause(dag_id=dag_id, before=before)
+    delete_stmt = delete(DatasetDagRunQueue).where(*where_clause)
+    result = session.execute(delete_stmt)
+    if result.rowcount > 0:
+        return NoContent, HTTPStatus.NO_CONTENT
+
+    raise NotFound(
+        "Queue event not found",
+        detail=f"Queue event with dag_id: `{dag_id}` was not found",
+    )
+
+
+@security.requires_access_dataset("GET")
```

Comment:
We probably need to filter the records we return to only DAGs that the user has read on.

Reply:
Sure. Just added it!